### PR TITLE
add vehicle on parking detection models

### PR DESCRIPTION
**Pull request by ygnn123**

### CODE REVIEW

File: pytorch_toolkit/object_detection/vehicle-detection/readme.md [hunk 1]
```diff
@@ -5,6 +5,10 @@
 | vehicle-detection-0200 | 0.82 | 1.83 | 25.4 | [snapshot](https://download.01.org/opencv/openvino_training_extensions/models/object_detection/v2/vehicle-detection-0200-1.pth), [configuration file](./vehicle-detection-0200/config.py) | 4 |
 | vehicle-detection-0201 | 1.84 | 1.83 | 32.3 | [snapshot](https://download.01.org/opencv/openvino_training_extensions/models/object_detection/v2/vehicle-detection-0201-1.pth), [configuration file](./vehicle-detection-0201/config.py) | 4 |
 | vehicle-detection-0202 | 3.28 | 1.83 | 36.9 | [snapshot](https://download.01.org/opencv/openvino_training_extensions/models/object_detection/v2/vehicle-detection-0202-1.pth), [configuration file](./vehicle-detection-0202/config.py) | 4 |
+| vehicle-detection-0001 | 212.22 | 36.56 | 32.4 | [snapshot](https://download.01.org/opencv/openvino_training_extensions/models/object_detection/v2/vehicle-detection-0203.pth), [configuration file](./vehicle-detection-0203/config.py) | 4 |
```
Comment (AlexanderDokuchaev) [inline pytorch_toolkit/object_detection/vehicle-detection/readme.md:8]: Names of models should be 0203,  0204, 0205

File: pytorch_toolkit/object_detection/vehicle-detection/readme.md [hunk 2]
```diff
@@ -101,7 +105,7 @@ python ../../external/mmdetection/tools/export.py \
 This produces model `$MODEL_NAME.xml` and weights `$MODEL_NAME.bin` in single-precision floating-point format
 (FP32). The obtained model expects **normalized image** in planar BGR format.
 
-For SSD networks an alternative OpenVINO™ representation is possible.
+For SSD networks (0200,0201,0201) an alternative OpenVINO™ representation is possible.
```
Comment (AlexanderDokuchaev) [inline pytorch_toolkit/object_detection/vehicle-detection/readme.md:108]: Missed 0202

**Comment by AlexanderDokuchaev** [comment]
run tests

**Comment by Ilya-Krylov** [comment]
run tests

**Comment by Ilya-Krylov** [comment]
@ygnn123 There were some difficulties with export. Have you resolve them? We need to have tests on export and testing

**Comment by ygnn123** [comment]
There are not problems now.

File: pytorch_toolkit/object_detection/vehicle-detection/readme.md [hunk 3]
```diff
@@ -5,6 +5,10 @@
 | vehicle-detection-0200 | 0.82 | 1.83 | 25.4 | [snapshot](https://download.01.org/opencv/openvino_training_extensions/models/object_detection/v2/vehicle-detection-0200-1.pth), [configuration file](./vehicle-detection-0200/config.py) | 4 |
 | vehicle-detection-0201 | 1.84 | 1.83 | 32.3 | [snapshot](https://download.01.org/opencv/openvino_training_extensions/models/object_detection/v2/vehicle-detection-0201-1.pth), [configuration file](./vehicle-detection-0201/config.py) | 4 |
 | vehicle-detection-0202 | 3.28 | 1.83 | 36.9 | [snapshot](https://download.01.org/opencv/openvino_training_extensions/models/object_detection/v2/vehicle-detection-0202-1.pth), [configuration file](./vehicle-detection-0202/config.py) | 4 |
+| vehicle-detection-0003 | 212.22 | 36.56 | 32.4 | [snapshot](https://download.01.org/opencv/openvino_training_extensions/models/object_detection/v2/vehicle-detection-0203.pth), [configuration file](./vehicle-detection-0203/config.py) | 4 |
```
Comment (Ilya-Krylov) [inline pytorch_toolkit/object_detection/vehicle-detection/readme.md:8]: @ygnn123 Will you fix these numbers?

**Comment by Ilya-Krylov** [comment]
@ygnn123 please resolve conflicts and prepare template.yaml files

**Comment by Ilya-Krylov** [comment]
@ygnn123 could you please bring approved test data (5-10 pictures) and expected quality metric value on it? Corresponding test is also needed (train, eval, export). Please have a look at `pytorch_toolkit/object_detection/tests/` folder.

**Comment by ygnn123** [comment]
> @ygnn123 could you please bring approved test data (5-10 pictures) and expected quality metric value on it? Corresponding test is also needed (train, eval, export). Please have a look at `pytorch_toolkit/object_detection/tests/` folder.

@Ilya-Krylov , good day!

CI fails during export step with error:
  File "../../../tools/export.py", line 42, in <module>
      if Config.fromfile(yaml.load(read_file, yaml.SafeLoader)['config']).model.bbox_head.type == 'SSDHead'
because CascadeRCNN config does not have bbox_head under model. It is under roi_head. Look here:
https://github.com/ygnn123/openvino_training_extensions/blob/yg/add_veh_park_det/pytorch_toolkit/object_detection/model_templates/vehicle-detection/vehicle-detection-0203/config.py

File: .gitmodules [hunk 4]
```diff
@@ -7,7 +7,7 @@
 	url = https://github.com/tensorflow/models.git
 [submodule "external/mmdetection"]
 	path = external/mmdetection
-	url = ../mmdetection
+	url = https://github.com/openvinotoolkit/mmdetection.git
```
Comment (Ilya-Krylov) [inline .gitmodules:10]: why? please use ../mmdetection

**Comment by Ilya-Krylov** [comment]
@ygnn123 You also have some problems with documentation http://134.191.240.124/job/ote/job/precommit/job/documentation/725/console

File: pytorch_toolkit/object_detection/model_templates/vehicle-detection/readme.md [hunk 5]
```diff
@@ -7,7 +7,8 @@ Models that are able to detect vehicles.
...
 | vehicle-detection-0201 | 1.84 | 1.83 | 32.3 | [snapshot](https://download.01.org/opencv/openvino_training_extensions/models/object_detection/v2/vehicle-detection-0201-1.pth), [model template](./vehicle-detection-0201/template.yaml) | 4 |
 | vehicle-detection-0202 | 3.28 | 1.83 | 36.9 | [snapshot](https://download.01.org/opencv/openvino_training_extensions/models/object_detection/v2/vehicle-detection-0202-1.pth), [model template](./vehicle-detection-0202/template.yaml) | 4 |
-
+| vehicle-detection-0003 | 112.34 | 24.11 | 43.8 | [snapshot](https://download.01.org/opencv/openvino_training_extensions/models/object_detection/v2/vehicle-detection-0203.pth), [model template](./vehicle-detection-0203/template.yaml)) | 4 |
+| vehicle-detection-0004 | 190.32 | 34.15 | 47.3 | [snapshot](https://download.01.org/opencv/openvino_training_extensions/models/object_detection/v2/vehicle-detection-0204.pth), [model template](./vehicle-detection-0204/template.yaml)) | 4 |
```
Comment (Ilya-Krylov) [inline pytorch_toolkit/object_detection/model_templates/vehicle-detection/readme.md:11]: add empty line after the table, otherwise it is shown wrong

**Comment by Ilya-Krylov** [comment]
Please revert 'Submodule deep-object-reid deleted from ddde34'

File: pytorch_toolkit/object_detection/tests/export_tests_vehicle_detection.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+class VehicleDetection0203TestCase(
+        create_export_test_case(
+            'vehicle-detection',
+            'vehicle-detection-0203',
+            os.path.dirname(__file__) + '/../../../data/vehicle_detection/annotation_val.json',
+            os.path.dirname(__file__) + '/../../../data/vehicle_detection/val',
+            False)
+):
+    """ Test case for vehicle-detection-0230 model export. """
```
Comment (Ilya-Krylov) [inline pytorch_toolkit/object_detection/tests/export_tests_vehicle_detection.py:61]: 203

File: pytorch_toolkit/object_detection/model_templates/vehicle-detection/readme.md [hunk 7]
```diff
@@ -7,7 +7,8 @@ Models that are able to detect vehicles.
 | vehicle-detection-0200 | 0.82 | 1.83 | 25.4 | [snapshot](https://download.01.org/opencv/openvino_training_extensions/models/object_detection/v2/vehicle-detection-0200-1.pth), [model template](./vehicle-detection-0200/template.yaml) | 4 |
 | vehicle-detection-0201 | 1.84 | 1.83 | 32.3 | [snapshot](https://download.01.org/opencv/openvino_training_extensions/models/object_detection/v2/vehicle-detection-0201-1.pth), [model template](./vehicle-detection-0201/template.yaml) | 4 |
 | vehicle-detection-0202 | 3.28 | 1.83 | 36.9 | [snapshot](https://download.01.org/opencv/openvino_training_extensions/models/object_detection/v2/vehicle-detection-0202-1.pth), [model template](./vehicle-detection-0202/template.yaml) | 4 |
-
+| vehicle-detection-0003 | 112.34 | 24.11 | 43.8 | [snapshot](https://download.01.org/opencv/openvino_training_extensions/models/object_detection/v2/vehicle-detection-0203.pth), [model template](./vehicle-detection-0203/template.yaml)) | 4 |
```
Comment (Ilya-Krylov) [inline pytorch_toolkit/object_detection/model_templates/vehicle-detection/readme.md:10]: 203

File: pytorch_toolkit/object_detection/model_templates/vehicle-detection/readme.md (hunk 5 above)
Comment (Ilya-Krylov) [inline pytorch_toolkit/object_detection/model_templates/vehicle-detection/readme.md:11]: 204

**Review by Ilya-Krylov — CHANGES_REQUESTED** [review]

File: .gitmodules [hunk 8]
```diff
@@ -7,13 +7,13 @@
...
 [submodule "external/mmaction"]
 	path = external/mmaction
 	url = ../mmaction.git
 [submodule "external/deep-object-reid"]
 	path = external/deep-object-reid
-	url = ../deep-object-reid.git
+	url = https://github.com/openvinotoolkit/deep-object-reid.git
```
Comment (Ilya-Krylov) [inline .gitmodules:16]: Relative paths!

**Comment by Ilya-Krylov** [comment]
merdged https://github.com/openvinotoolkit/training_extensions/pull/370

**Pull request closed without merging**